### DISCUSSION TITLE
theia can run springboot, but not debug springboot, why?

### DISCUSSION THREAD

**Discussion by aaa0long** [eclipse-theia/theia#10332] [Q&A]
### Bug Description:
theia cannot debug springboot，but can run springboot。
Error：
org.eclipse.jdi.internal.connect.ConnectorImpl$StringArgumentImpl.<init>(Lorg/eclipse/jdi/internal/connect/ConnectorImpl;Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;Z)V

### Steps to Reproduce:

1.
![Snipaste_2021-10-27_11-52-03](https://user-images.githubusercontent.com/86939787/138997452-4ed532f6-59fb-44be-ac1c-59834780648f.png)

2.

![Snipaste_2021-10-27_11-47-10](https://user-images.githubusercontent.com/86939787/138997525-70b106f5-374f-410c-8f0d-d9a96a82e839.png)

3.

### Additional Information

- Operating System: inux ubuntu 5.11.0-38-generic #42~20.04.1-Ubuntu SMP Tue Sep 28 20:41:07 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
- Theia Version:  1.17.2 (Beta)   VS Code API Version: 1.50.0
- jdk : 
![Snipaste_2021-10-27_11-48-40](https://user-images.githubusercontent.com/86939787/138997844-75fb5bd0-dc8f-4c46-97c4-c6ddbb28e658.png)
- theia dependency: 
![Snipaste_2021-10-27_11-47-54](https://user-images.githubusercontent.com/86939787/138997906-951bfaea-2d91-42eb-a945-5d21b766831a.png)

**Comment by aaa0long** [comment]
please help me！！！

**Comment by msujew** [comment]
Hi @aaa0long, thanks for your question.

I'm not that knowledgeable in regards to Java extension support, but I imagine that the spring extensions don't actually provide a debugger. However, there are other extensions like the [Debugger for Java](https://open-vsx.org/extension/vscjava/vscode-java-debug) which should provide such a service. You should install that in conjunction with the [Redhat Java](https://open-vsx.org/extension/redhat/java) extension.

**Reply by aaa0long** [reply]
Hi msujew, gitpod can run and debug springboot,  but theia can not debug.  Has anyone practiced this?  Can theia debug the program, or is my step wrong?

**Comment by aaa0long** [comment]
Hi @msujew, thanks for your reply.
I'm have dependency about Debugger for Java，but still have that error. I'm going crazy.
Error:
org.eclipse.jdi.internal.connect.ConnectorImpl$StringArgumentImpl.(Lorg/eclipse/jdi/internal/connect/ConnectorImpl;Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;Z)V


![Snipaste_2021-10-27_18-00-49](https://user-images.githubusercontent.com/86939787/139045092-5b09821f-d932-4026-aa6f-1785178cc318.png)

**Comment by msujew** [comment]
@tsmaeder You probably have a bit more experience in that field. Any idea what could be wrong here?

**Comment by tsmaeder** [comment]
That sounds more like a question for the folks writing the debugger extension.
@aaa0long do you have more of a stack trace than just 
"org.eclipse.jdi.internal.connect.ConnectorImpl$StringArgumentImpl.(Lorg/eclipse/jdi/internal/connect/ConnectorImpl;Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;Z)V"?

In particular, what's the exception and complete stack trace?

**Reply by aaa0long** [reply]
Hi @tsmaeder  thanks for your reply.
Stack Trace：
1. 
![Snipaste_2021-10-27_18-00-49](https://user-images.githubusercontent.com/86939787/139173611-ab53a324-860e-49ed-9160-3df1ce1559a3.png)
2.
![Snipaste_2021-10-28_09-56-52](https://user-images.githubusercontent.com/86939787/139173706-43be2e0c-5df3-48fc-a150-bdf400233d69.png)
